File: 2025-1/sesion-de-clase-17/SoftProgNet/SoftProgNegocio/BOImpl/ClienteBOImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PUCP.Edu.Pe.SoftProg.Modelo.Clientes;
using PUCP.Edu.Pe.SoftProg.Negocio.BO;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAO.Clientes;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAOImpl.Clientes;

namespace PUCP.Edu.Pe.SoftProg.Negocio.BOImpl {
    public class ClienteBOImpl : IClienteBO {
        private readonly IClienteDAO clienteDAO;

        public ClienteBOImpl() {
            this.clienteDAO = new ClienteDAOImpl();
        }

        public void Guardar(Cliente cliente, Estado estado) {
            if (estado == Estado.Nuevo) {
                this.clienteDAO.Insertar(cliente);
            }
            else {
                this.clienteDAO.Modificar(cliente);
            }
        }

        public void Eliminar(int id) {
            this.clienteDAO.Eliminar(id);
        }

        public Cliente Obtener(int id) {
            return this.clienteDAO.Buscar(id);
        }

        public List<Cliente> Listar() {
            return this.clienteDAO.Listar();
        }

        public Cliente BuscarPorDni(string dni)
        {
            return this.clienteDAO.BuscarPorDni(dni);
        }
    }
}

File: 2025-1/sesion-de-clase-20/SoftProg/SoftProgNegocio/src/main/java/pe/edu/pucp/inf30/softprog/boimpl/ClienteBOImpl.java
package pe.edu.pucp.inf30.softprog.boimpl;

import java.util.List;
import pe.edu.pucp.inf30.softprog.bo.IClienteBO;
import pe.edu.pucp.inf30.softprog.dao.clientes.IClienteDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.clientes.ClienteDAOImpl;
import pe.edu.pucp.inf30.softprog.model.Estado;
import pe.edu.pucp.inf30.softprog.model.clientes.Cliente;

/**
 *
 * @author eric
 */
public class ClienteBOImpl implements IClienteBO {
    private final IClienteDAO clienteDao;
    
    public ClienteBOImpl() {
        this.clienteDao = new ClienteDAOImpl();
    }
    
    @Override
    public Cliente buscarPorDni(String dni) {
        return this.clienteDao.buscarPorDni(dni);
    }

    @Override
    public List<Cliente> listar() {
        return this.clienteDao.listar();
    }

    @Override
    public Cliente obtener(int id) {
        return this.clienteDao.buscar(id);
    }

    @Override
    public void eliminar(int id) {
        this.clienteDao.eliminar(id);
    }

    @Override
    public void guardar(Cliente modelo, Estado estado) {
        if (estado == Estado.Nuevo) {
            this.clienteDao.insertar(modelo);
        }
        else {
            this.clienteDao.modificar(modelo);
        }
    }
}
